### PR TITLE
feat(TFD-4542): Upgrade to Beam 2.5

### PR DESCRIPTION
**Pull request by bchen-talend**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
Upgrade Beam version to 2.5

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/154/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/154/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/155/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/155/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/161/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/161/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/162/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/162/)

File: components-parent/components-bom/pom.xml [hunk 1]
```diff
@@ -122,7 +122,7 @@
         <json-simple.version>1.1.1</json-simple.version>
         <jsr305.version>1.3.9</jsr305.version>
         <junit.version>4.12</junit.version>
-        <kafka-clients.version>0.9.0.1</kafka-clients.version>
+        <kafka-clients.version>1.0.0</kafka-clients.version>
```
Comment (amarouni) [inline components-parent/components-bom/pom.xml]: Is this the kafka client version for the KafkaIO component ? 
If that's the case, is it still compatible with Kafka < 1.0.0 ?

Comment (bchen-talend) [reply components-parent/components-bom/pom.xml]: good point, from the API doc yes, and tested with  0.11 on our stack and 0.10 on IT, but not 0.9
and another point is on kafka connection property there is version which is not really useful as KafkaIO provide the compatibility, can be removed

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/28/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/28/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Abchen%2FTFD-4542-Beam25)

**Pull request merged**